README.md

```diff
@@ -1,3 +1,267 @@
+environment variables, and command line flags
+
+``` go
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/vimeo/dials/ez"
+)
+
+// Config is the configuration struct needed for the application
+type Config struct {
+	// When a struct tag more specifically corresponding to a source is
+	// present, it takes precedence over the `dials` tag. Note that just
+	// because there is a `yaml` struct tag present doesn't mean that other
+	// sources can't fill this field.
+	Val1 string `dials:"Val1" yaml:"b"`
+	// the dials tag can be used as alias so when the name in the config file
+	// changes, the code doesn't have to change.
+	Val2 int `dials:"val_2"`
+	// Dials follows the Go convention for flags and will look for the dials
+	// tag or field name in lower-kebab-case. In this case, it would look for
+	// val-3 flag. To specify a different flag name, use the `dialsflag` tag.
+	// Now, Dials will lookup "some-val" flag instead. Since only
+	// `dialsflag` tag is specified, Val3 will only be populated from command
+	// line flags
+	Val3 bool `dialsflag:"some-val"`
+	// Path holds the value of the path to the config file. Dials follows the
+	// Go convention and will look for the dials tag or field name in all caps. 
+	// In this case, it would lookup the PATH environment variable. To specify
+	// a different env variable, use the `dials_env` tag. Now Dials will lookup
+	// "configpath" env value to populate the Path field
+	Path string `dials_env:"configpath"`
+}
+
+// ConfigPath returns the path to the config file. This is particularly helpful
+// when the path is populated from environment variables or command line flags.
+// Dials will first read from environment variables and command line flags
+// and then read the config file specified by the populated field.
+func (c *Config) ConfigPath() (string, bool) {
+	// can alternatively return empty string and false if the state of the
+	// struct doesn't specify a config file to read
+	return c.Path, true
+}
+
+func main() {
+	c := &Config{}
+
+	// The following function will populate the config struct by reading the
+	// config files, environment variables, and command line flags (order matches
+	// the function name) with increasing precedence. In other words, the flag
+	// source (last) would overwrite the YAML source (first) were they both to
+	// attempt to set the same struct field. The boolean argument passed to the
+	// function indicates whether the file will be watched and updates to the
+	// file should update the config struct.
+	view, dialsErr := ez.YAMLConfigEnvFlag(context.Background(), c, false)
+	if dialsErr != nil {
+		// error handling
+	}
+
+	// Get an interface corresponding to the filled-out config struct, and
+	// assert it to the correct type. Here's the struct populated from config file,
+	// environment variables, and command line flags.
+	Config := view.View().(*Config)
```

Comment:
A bit strange to have a variable with an initial capital letter; maybe just call this `config`.

Also, it might be nice to show an example of the `Fill` methods for those that might be put off by the necessary type assert (until we get Generics!):
```go
config := &Config{}
d.Fill(config)
```